packages/trader/src/App/Components/Elements/ContractAudit/contract-details.tsx

```diff
@@ -16,17 +15,41 @@ import {
     isEndedBeforeCancellationExpired,
     isUserCancelled,
     toGMTFormat,
+    TContractInfo,
 } from '@deriv/shared';
 import {
     addCommaToNumber,
     getBarrierLabel,
     getBarrierValue,
     isDigitType,
 } from 'App/Components/Elements/PositionsDrawer/helpers';
-import ContractAuditItem from './contract-audit-item.jsx';
+import ContractAuditItem from './contract-audit-item';
 import { isCancellationExpired } from 'Stores/Modules/Trading/Helpers/logic';
 
-const ContractDetails = ({ contract_end_time, contract_info, duration, duration_unit, exit_spot, is_vanilla }) => {
+type TTContractInfoWithCancellation = TContractInfo & {
+    cancellation: {
```

Comment:
@kate-deriv as I can see, `TContractInfo` type already includes correct `cancellation` property. please elaborate why we're overriding it here? :)

Reply:
please use the correct `TContractInfo` type for all `contract_info` agruments in `logic.ts` instead of `TIsEndedBeforeCancellationExpired` and `TGetEndTime` types 🙏🏻 , and you won't have to extend TContractInfo here and in the test file for this component :)

Reply:
after replacing TTContractInfoWithCancellation with TContractInfo in this file, you will also need to wrap `exit_tick_time` in to Number() :)

Reply:
So I decided to do the same

Reply:
yes, `TIsEndedBeforeCancellationExpired`  is a wrong type. We need to replace it with `TContractInfo`, those obligatory properties are not obligatory at all. After you replace this wrong type with `TContractInfo` in logic.ts, there will be just one line in that file that will need you to add a '?', that's it. It's worth it because we will have the correct type. We should get rid of the wrong type that re-invents the wheel which was added by someone long time ago just to satisfy TS. I had to replace some wrong temporary types in Turbos PR earlier because we have to own this. :)